platform/cog-platform-fdo.c

```diff
@@ -340,6 +355,68 @@ static const struct zxdg_toplevel_v6_listener xdg_toplevel_listener = {
     .close = xdg_toplevel_on_close,
 };
 
+
+static void noop()
+{
+  // This space intentionally left blank
+}
+
+static void
+output_handle_scale (void *data,
+                     struct wl_output *output,
+                     int32_t factor)
+{
+    bool found = false;
+    for (int i = 0; i < 16; i++)
+    {
+        if (wl_data.metrics[i].output == output)
+        {
+            found = true;
+            wl_data.metrics[i].scale = factor;
+            break;
+        }
+    }
+    if (!found)
+    {
+        printf("Unknown output %p\n", output);
```

Comment:
Please leave a space between the function name and the opening parenthesis for the argument list:

```c
printf ("Unknown output %p\n", output);
```

This needs to be fixed in a few places more, but I'll comment it here just once.